Backport of PHP 8.5 patch to 5.x.x

reedy opened the pull request:
ci: Add PHP 8.5 to pipeline, ignoring dependencies and as experimental (#842)

This PR will
- resolve deprecations for:
  - Reflection*::setAccessible
  - curl_close

Fixes #798
Fixes #863

DannyvdSluijs committed:
fix: Upgrade php cs fixer to latest (#783)

Fixes #598

DannyvdSluijs committed:
Backport of PHP 8.5 patch:

ci: Add PHP 8.5 to pipeline, ignoring dependencies and as experimental (#842)

This PR will
- resolve deprecations for:
  - Reflection*::setAccessible
  - curl_close

Fixes #798
Fixes #863

gnutix committed:
Fix PHP 8.5 $http_response_header deprecation on 5.x.x (#857)

As requested in
https://github.com/jsonrainbow/json-schema/pull/841#issuecomment-3607560578

DannyvdSluijs commented:
@reedy thanks for the PR and I’m willingly get this merged into 5.x.x. This might take some while as I’m enjoying the holidays atm. 

Looks like the CI pipeline indeed doesn’t work so I’ll have to jump on that before we merge things. Do you already found something or have a idea on why it isn’t running them?

reedy commented:
I'm pretty sure it's because there's no `.github/workflow` files in the branch, and I'm not trying to merge into `main`, so it doesn't know to fallback to anywhere. I haven't seen any docs suggest that there's any sort of fallback behaviour (at least by default, you may be able to define it somehow)

https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#onpull_requestpull_request_targetbranchesbranches-ignore

>When using the pull_request and pull_request_target events, you can configure a workflow to run only for pull requests that target specific branches.

The easiest answer is probably copying a version of the relevant CI files you want to run back into that branch, and it'll probably work fine... I guess that should be easily proved, will do a patch.

No rush on my part, I'm just a little bored, and upgrading an ancient version of composer (while running on PHP 8.5) locally noticed another PHP 8.5 cleanup that needed doing in MediaWiki and various vendored libs, such as this one :)

reedy committed:
Backport some CI to 5.x.x

Remove old .travis.yml

Update .gitattributes

cf https://github.com/jsonrainbow/json-schema/tree/2911a17/.github/workflows
